src/main/scala/workflow/AutoCacheRule.scala

```diff
@@ -423,19 +434,20 @@ class AutoCacheRule(cachingMode: CachingStrategy) extends Rule with Logging {
    * Returns true iff there is still an uncached node whose output is used > once, that would fit in memory
    * if cached
    */
-  def stillRoom(caches: Set[Int], runs: Map[Int, Int], profiles: Map[Int, Profile], spaceLeft: Long): Boolean = {
-    runs.exists { i =>
-      (i._2 > 1) &&
-        (!caches.contains(i._1)) &&
-        (profiles.getOrElse(i._1, Profile(0, 0, 0)).rddMem < spaceLeft)
+  def stillRoom(caches: Set[Int], runs: Array[Int], profiles: Map[Int, Profile], spaceLeft: Long): Boolean = {
```

Comment:
It didn't buy much if anything in terms of performance, but I was changing `getRuns` anyway, so I figured I may as well change that too just in case.